src/features/TodosViewForm.jsx

```diff
@@ -1,5 +1,16 @@
+import { useEffect, useState } from 'react';
+
 function TodosViewForm({ sortDirection, setSortDirection, sortField, setSortField, queryString, setQueryString }) {
 
+  const [localQueryString, setLocalQueryString] = useState(queryString);
+
+  useEffect(() => {
+    const debounce = setTimeout(() => {
+      setQueryString(localQueryString)
+    }, 500);
+    return () => { clearInterval(debounce) };
```

Comment:
This could lead to unexpected bugs since we are dealing with a "time out" and not "interval".` clearTimeout` instead of `clearInterval` is a better approach.